Tests/SemanticVersioningTests/SemanticVersionTests.swift

```diff
@@ -24,37 +24,47 @@ class SemanticVersionTests: XCTestCase {
     }
     
     func testOrder() async throws {
-        //  1.0.0 < 2.0.0 < 2.1.0 < 2.1.1
-        let v100 = SemanticVersion(
+        //  1.2.0 < 2.0.0 < 2.0.1 < 2.1.0
```

Comment:
This version cases from https://semver.org .
So please make new test case for this issue.

Reply:
@noppefoxwolf 
I see, do you also prefer to split `testPreReleasedOrder`?

Reply:
No need to that. 👍